pkg/controller/nodes/transformers.go

```diff
@@ -94,11 +94,24 @@ func ToNodeExecutionEvent(nodeExecID *core.NodeExecutionIdentifier,
 		phase = core.NodeExecution_RUNNING
 	}
 
-	nev := &event.NodeExecutionEvent{
-		Id:         nodeExecID,
-		Phase:      phase,
-		InputUri:   inputPath,
-		OccurredAt: occurredTime,
+	var nev *event.NodeExecutionEvent
+	if nodeExecID.NodeId == v1alpha1.StartNodeID {
+		nev = &event.NodeExecutionEvent{
+			Id:       nodeExecID,
+			Phase:    phase,
+			InputUri: v1alpha1.GetOutputsFile(status.GetOutputDir()).String(),
```

Comment:
move the `v1alpha1.GetOutputsFile(status.GetOutputDir()).String()` to a common var in the if block? This will make it clear that outputs are being copied to both inputs and outputs. Also, can you write a comment about this?

@katrogan, if we send the entire data within the payload, would the input and output uris be unfurled each time. If so should we move this peculiar logic to Admin?

Reply:
So right the OutputUri is not even sent for the start-node so bare minimum we would need a change to send that from propeller. And also that would be similar custom logic which checks for the start-node and populates the OutputUri or InputUri in propeller and then admin can just copy over the value with another custom logic there for the start-node to other field.

Reply:
@kumare3 if we send the URIs in the payload admin will not unfurl them unless someone calls for GetNodeExecutionData - which itself perhaps ought to be (lazily) cached somehow

Reply:
@katrogan i was referring to https://github.com/flyteorg/flytepropeller/blob/61893990bb8d0a46133ce0cc81b94f9c1e647764/pkg/controller/events/node_execution.go#L53

Reply:
@kumare3 we can't always unfurl in admin, right? because the whole purpose of setting the policy at propeller side is to support a distributed propeller with protected s3 access